src/main/java/com/thullo/service/BoardServiceImpl.java

```diff
@@ -73,15 +70,16 @@ private BoardResponse getBoard(Board board) {
         return getBoardResponse(board);
     }
 
-    private Board getBoardInternal(String boardTag) {
-        return boardRepository.findByBoardTag(boardTag).orElse(null);
+    public Board getBoardInternal(String boardTag) throws BadRequestException {
```

Comment:
The method is not meant to throw an exception, kindly revert back to the previous implementation